linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java

```diff
@@ -0,0 +1,193 @@
+      CommonVars$.MODULE$.apply(
+          "linkis.mysql.force.params",
+          "allowLoadLocalInfile=false&autoDeserialize=false&allowLocalInfile=false&allowUrlInLocalInfile=false");
+
+  public static final CommonVars<String> MYSQL_STRONG_SECURITY_ENABLE =
+      CommonVars$.MODULE$.apply("linkis.mysql.strong.security.enable", "false");
+
+  /**
+   * mysql url append force params
+   *
+   * @param url
+   * @return
+   */
+  public static String appendMysqlForceParams(String url) {
+    if (StringUtils.isBlank(url)) {
+      return "";
+    }
+    Map<String, Object> forceParams = parseMysqlUrlParamsToMap(MYSQL_FORCE_PARAMS.getValue());
+
+    String extraParamString = parseParamsMapToMysqlParamUrl(forceParams);
+
+    if (url.endsWith(QUESTION_MARK)) {
+      url = url + extraParamString;
+    } else if (url.lastIndexOf(QUESTION_MARK) < 0) {
+      url = url + QUESTION_MARK + extraParamString;
+    } else {
+      url = url + AND_SYMBOL + extraParamString;
+    }
+    return url;
+  }
+
+  public static void appendMysqlForceParams(Map<String, Object> extraParams) {
+    extraParams.putAll(parseMysqlUrlParamsToMap(MYSQL_FORCE_PARAMS.getValue()));
+  }
+
+  public static String checkJdbcSecurity(String url) {
+    logger.info("checkJdbcSecurity origin url: {}", url);
+    if (StringUtils.isBlank(url)) {
+      throw new LinkisSecurityException(35000, "Invalid mysql connection cul, url is empty");
+    }
+    if (url.endsWith(QUESTION_MARK) || !url.contains(QUESTION_MARK)) {
+      logger.info("checkJdbcSecurity target url: {}", url);
+      return url;
+    }
+    String[] items = url.split("\\?");
+    if (items.length != 2) {
+      logger.warn("Invalid url: {}", url);
+      throw new LinkisSecurityException(35000, "Invalid mysql connection cul: " + url);
+    }
+    Map<String, Object> params = parseMysqlUrlParamsToMap(items[1]);
+    Map<String, Object> securityMap = checkJdbcSecurity(params);
+    String paramUrl = parseParamsMapToMysqlParamUrl(securityMap);
+    url = items[0] + QUESTION_MARK + paramUrl;
+    logger.info("checkJdbcSecurity target url: {}", url);
+    return url;
+  }
+
+  /**
+   * check jdbc params
+   *
+   * @param paramsMap
+   */
+  public static Map<String, Object> checkJdbcSecurity(Map<String, Object> paramsMap) {
+    if (paramsMap == null) {
+      return new HashMap<>();
+    }
+
+    // mysql url strong security
+    if (Boolean.valueOf(MYSQL_STRONG_SECURITY_ENABLE.getValue())) {
+      paramsMap.clear();
+      return paramsMap;
+    }
+
+    Iterator<Map.Entry<String, Object>> iterator = paramsMap.entrySet().iterator();
+    while (iterator.hasNext()) {
+      Map.Entry<String, Object> entry = iterator.next();
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (StringUtils.isBlank(key) || value == null || StringUtils.isBlank(value.toString())) {
+        logger.warn("Invalid parameter key or value is blank.");
+        iterator.remove();
```

Comment:
Why remove?

Reply:
Get rid of arguments where key or value is blank.